src/components/birth-date-picker/birth-date-picker.tsx

```diff
@@ -0,0 +1,172 @@
+import React = require("react");
+import {autorun, computed, observable, reaction} from "mobx";
+import {observer} from "mobx-react";
+
+function isValidDate(date: any): boolean {
+    return date instanceof Date && !Number.isNaN(date.getTime());
```

Comment:
since you are not exporting this function, why not just give it a type and then you don't need to check that the type is date